mm2src/mm2_rpc/src/data/legacy.rs

```diff
@@ -254,4 +256,95 @@ pub struct MmVersionResponse {
     pub datetime: String,
 }
 
-fn get_true() -> bool { true }
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(tag = "method", rename_all = "lowercase")]
+pub enum ActivationRequest {
+    Enable(EnableRequest),
+    Electrum(ElectrumRequest),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct EnableRequest {
+    pub coin: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub urls: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub swap_contract_address: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fallback_swap_contract: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_url: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_decimals: Option<u8>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_policy: Option<GasStationPricePolicy>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mm2: Option<u8>,
+    #[serde(default)]
+    pub tx_history: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_confirmations: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub requires_notarization: Option<bool>,
+}
+
+/// Using tagged representation to allow adding variants with coefficients, percentage, etc in the future.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "policy", content = "additional_data")]
+pub enum GasStationPricePolicy {
+    /// Use mean between average and fast values, default and recommended to use on ETH mainnet due to
+    /// gas price big spikes.
+    MeanAverageFast,
+    /// Use average value only. Useful for non-heavily congested networks (Matic, etc.)
+    Average,
+}
+
+impl Default for GasStationPricePolicy {
+    fn default() -> Self { GasStationPricePolicy::MeanAverageFast }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ElectrumRequest {
+    pub coin: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub servers: Vec<Server>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mm2: Option<u8>,
+    #[serde(default)]
+    pub tx_history: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_confirmations: Option<u64>,
+    #[serde(default)]
+    pub requires_notarization: bool,
```

Comment:
`requires_notarization` defaults to coin config parameter not false.

Reply:
According to [the documentation](https://developers.komodoplatform.com/basic-docs/atomicdex-api-legacy/coin_activation.html#electrum) - `requires_notarization`'s default is false. 
Should we bring documentation in order?

cc: @smk762

Reply:
I believe it does default to false if not in coins config, as notarisation only applies to a subset of coins. We can clarify this in the docs.
So the hierarchy is:
1. Method param value
2. Activation param value
3. Coins config value
4. False

Reply:
Thank you for the fast feedback!
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1912/commits/4dc23e6d9c9d8a20427a5827471bf23674e3a613)